windows/src/developer/TIKE/actions/dmActionsMain.pas

```diff
@@ -560,13 +592,27 @@ procedure TmodActionsMain.OpenProject(FileName: WideString);
 
 procedure TmodActionsMain.NewProject(pt: TProjectType);
 begin
-  FGlobalProject.Save;
-  FreeGlobalProjectUI;
+  if IsGlobalProjectUIReady then
+  begin
+    if not SaveAndCloseAllFiles then Exit;
+    FreeGlobalProjectUI;
+  end;
   NewGlobalProjectUI(pt);
   frmKeymanDeveloper.ShowProject;
   frmKeymanDeveloper.UpdateCaption;
 end;
 
+procedure TmodActionsMain.CloseProject;
+begin
+  if IsGlobalProjectUIReady then
+  begin
+    if not SaveAndCloseAllFiles then Exit;
+  end;
+  FreeGlobalProjectUI;
```

Comment:
Is it safer to move `FreeGlobalProjectUI` inside the `if` block as done in ll. 585 and 598?

Reply:
It won't matter because it will be a no-op but will move for consistency :)